services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/outputs/_manager.py

```diff
@@ -171,14 +172,22 @@ def _remove_downloads(future: Future) -> None:
                 except Exception as e:  # pylint: disable=broad-except
                     self._last_upload_error_tracker[port_key] = e
 
```

Comment:
Q: I do not know all the context but can it happen that `self._task_uploading_followup is not None` at this point, i.e. is already holding a previously fired task? You specifically do that check  afterwards in the code `if self._task_uploading_followup is not None:`